### PR TITLE
feat(CORSO-41)Add a FAQ

### PR DESCRIPTION
**Pull request by DenisaNicoletaH**
JIRA:https://champlainsaintlambert.atlassian.net/browse/CORSO-41
## Context:
Allow admins to add a new FAQ 
## Changes

- Over 90% testing coverage
- Implemented a new AddAFAQ page
- 2 end-to-end testing
- Added caching to FAQ

## Before and After UI (Required for UI-impacting PRs)

Before: ** Page didn't exist'

After: 
image

### CODE REVIEW

File: corso_website_fe/src/Pages/Dashboard/ManageFaq/AddAFaq.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import pen from "../../../ressources/images/pen.svg";
+import {useAuth} from "../../../security/Components/AuthProvider";
+
+
+function AddAFaq() {
+    const [question, setQuestion] =  useState('');
+    const [answer, setAnswer] =  useState('');
+    // const [preference] =  useState('');
+
+    //const [faqid, setFaqid] = useState<FAQ>('');
```
Comment (KarinaSofia) [inline corso_website_fe/src/Pages/Dashboard/ManageFaq/AddAFaq.tsx:16]: Are these comments left for a reason?

File: corso_website_springboot/src/main/java/com/corso/springboot/FAQ_Subdomain/businesslayer/FAQService.java [hunk 2]
```diff
@@ -21,4 +23,10 @@ public interface FAQService {
     Integer getFAQCount();
 
     FAQResponse modifyFAQ(String faqId, FAQRequest faqRequest);
+
+
+//    FAQResponse createFAQ(FAQRequest faqRequest);
```
Comment (KarinaSofia) [inline corso_website_springboot/src/main/java/com/corso/springboot/FAQ_Subdomain/businesslayer/FAQService.java:28]: was this left by accident?

File: corso_website_springboot/src/main/resources/templates/ReviewPinnedChangeEmail.html [hunk 3]
```diff
@@ -23,26 +23,5 @@ <h2>Your review has been pinned by administrators.</h2>
...
-    <p>Pour plus d'informations sur votre commande, les rapports d'avancement ou un avis, veuillez vous connecter sur notre site web et nous contacter.</p>
-    <p>Nous sommes impatient de travailler avec vous!</p>
-    <hr>
-    <br>
-    <h1>Thank you for choosing Corso, </h1>
-    <h2>Your review has been unpinned by administrators.</h2>
-    <br>
-    <p>For more information about your order, progress reports or a review, please log in on our website and contact us.</p>
-    <p>We look forward to working with you!</p>
-</div>
```
Comment (KarinaSofia) [inline corso_website_springboot/src/main/resources/templates/ReviewPinnedChangeEmail.html:45]: why did you remove the french translation?

**Review by KarinaSofia — APPROVED** [review]
Great work denisa!!

File: corso_website_springboot/src/main/resources/templates/ReviewPinnedChangeEmail.html (hunk 3 above)
Comment (DenisaNicoletaH) [reply corso_website_springboot/src/main/resources/templates/ReviewPinnedChangeEmail.html:45]: because it was duplicated

File: corso_website_springboot/src/main/java/com/corso/springboot/FAQ_Subdomain/businesslayer/FAQService.java (hunk 2 above)
Comment (DenisaNicoletaH) [reply corso_website_springboot/src/main/java/com/corso/springboot/FAQ_Subdomain/businesslayer/FAQService.java:28]: I thought I might have needed it but in the end I didn't..

File: corso_website_fe/src/Pages/Dashboard/ManageFaq/AddAFaq.tsx (hunk 1 above)
Comment (DenisaNicoletaH) [reply corso_website_fe/src/Pages/Dashboard/ManageFaq/AddAFaq.tsx:16]: Yes I used them before but while I was trying out stuffs, I kept it for myself in case I needed it

**Review by Mila5847 — APPROVED** [review]
The comments left by our teammate are essentially what I wanted to ask, too. Other than that, looks great and it is tested, good!
